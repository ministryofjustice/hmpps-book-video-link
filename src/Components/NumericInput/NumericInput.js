import React from 'react'
import PropTypes from 'prop-types'

import ErrorText from '@govuk-react/error-text'
import Label from '@govuk-react/label'
import LabelText from '@govuk-react/label-text'
import HintText from '@govuk-react/hint-text'
import styled from 'styled-components'
import Input from '@govuk-react/input'

import { inputType, metaType } from '../../types'

const StyledInput = styled(Input)`
  width: 65px;
`
const NumericInput = ({ label, hint, meta, input }) => (
  <Label error={meta.error}>
    <LabelText> {label} </LabelText>
    {hint && <HintText> {hint} </HintText>}
    {meta.touched && meta.error && <ErrorText>{meta.error}</ErrorText>}
    <StyledInput type="number" error={meta.touched && meta.error} {...input} />
  </Label>
)

NumericInput.propTypes = {
  input: inputType.isRequired,
  meta: metaType,
  label: PropTypes.string.isRequired,
  hint: PropTypes.string,
}

NumericInput.defaultProps = {
  meta: {},
  hint: null,
}

export default NumericInput
