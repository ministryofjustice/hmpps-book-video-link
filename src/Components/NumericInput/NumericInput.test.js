import React from 'react'
import { shallow } from 'enzyme'
import NumericInput from './NumericInput'

describe('NumericInput', () => {
  it('should render correctly', () => {
    const input = {
      name: 'age',
      onChange: () => {},
    }
    const wrapper = shallow(<NumericInput input={input} label="Age" hint="Enter a value between 1 and 120" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with errors', () => {
    const input = {
      name: 'age',
      onChange: () => {},
    }
    const meta = {
      touched: true,
      error: 'Please enter an age between 1 and 120',
    }
    const wrapper = shallow(
      <NumericInput input={input} label="Age" hint="Enter a value between 1 and 120" meta={meta} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
