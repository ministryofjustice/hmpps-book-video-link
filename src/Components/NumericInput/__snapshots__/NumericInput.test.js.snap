// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInput should render correctly 1`] = `
<Label
  error={false}
>
  <LabelText>
     
    Age
     
  </LabelText>
  <HintText>
     
    Enter a value between 1 and 120
     
  </HintText>
  <Styled(Input)
    name="age"
    onChange={[Function]}
    type="number"
  />
</Label>
`;

exports[`NumericInput should render with errors 1`] = `
<Label
  error="Please enter an age between 1 and 120"
>
  <LabelText>
     
    Age
     
  </LabelText>
  <HintText>
     
    Enter a value between 1 and 120
     
  </HintText>
  <ErrorText>
    Please enter an age between 1 and 120
  </ErrorText>
  <Styled(Input)
    error="Please enter an age between 1 and 120"
    name="age"
    onChange={[Function]}
    type="number"
  />
</Label>
`;
